### PR TITLE
Feature/csh/final version lookup

### PR DESCRIPTION
**Pull request by cshcomcom**
## Description
- Add 'is_final' option to `readDatabase()`
- Add 'is_final' testcase

### CODE REVIEW

File: unittest/db.test.js [hunk 1]
```diff
@@ -282,6 +283,13 @@ describe("DB operations", () => {
         assert.deepEqual(node.db.getValue("/apps/test"), dbValues)
       })
 
+      it("when retrieving high value near top of database with is_final", () => {
+        const backupFinalVersion = node.db.stateManager.getFinalVersion();
+        node.db.stateManager.finalizeVersion(StateVersions.EMPTY);
+        assert.notDeepEqual(node.db.getValue("/apps/test", { isFinal: true }), dbValues)
```
Comment (platfowner) [inline unittest/db.test.js:289]: minor: what about just comparing with null?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks!

File: unittest/db.test.js (hunk 1 above)
Comment (cshcomcom) [reply unittest/db.test.js]: Looks good! Done

**Comment by cshcomcom** [comment]
Thanks for the review! Now merging

**Pull request merged**